### PR TITLE
Modify strategy to protect proxy

### PR DESCRIPTION
**Pull request by Vampouille**
This PR try to update strategy used to deny or allow proxy request. 
* Create a new configuration files in datadir : `proxy-permissions.xml` that define permissions of proxy request
* Add domain parameter to Permissions class
* Remove use of urlIsProtected() method

This Pr try to address issue when trying to use another website with same IP but another domain.

### CODE REVIEW

File: security-proxy/src/main/java/org/georchestra/security/Proxy.java [hunk 1]
```diff
@@ -205,8 +199,40 @@ public void init() throws Exception {
...
+
             logger.info("Done.");
         }
+
+        // Proxy permissions not set by datadir
+        if (proxyPermissionsFile != null && proxyPermissions == null) {
+            Closer closer = Closer.create();
+            try {
+                final ClassLoader classLoader = Proxy.class.getClassLoader();
+                InputStream inStream = closer.register(classLoader.getResourceAsStream(proxyPermissionsFile));
```
Comment (pmauduit) [inline security-proxy/src/main/java/org/georchestra/security/Proxy.java]: We should also be able to locate the permissions.xml file into the georchestra datadir

Comment (pmauduit) [reply security-proxy/src/main/java/org/georchestra/security/Proxy.java]: my mistake, it is done just above ...

**Comment by pmauduit** [comment]
Still need to avoid transmitting extrac `sec-*` headers when inappropriate, but should be good to go.

**Comment by Vampouille** [comment]
This commit 0b6cf45 remove `sec-*` header from proxy request.

**Comment by Vampouille** [comment]
Travis failed to run test on `ImpersonateUserRequestHeaderProviderTest` class but test run without errors on my workstation and I did not modify `ImpersonateUserRequestHeaderProvider` class. @pmauduit any idea ?

**Comment by pmauduit** [comment]
> @pmauduit any idea ?

Nope, I'll try on my dev env and try to reproduce, in the meantime, it could be worth trying to relaunch the build on travis.

**Comment by pmauduit** [comment]
> Nope, I'll try on my dev env and try to reproduce

Utests ok in eclipse & maven here

**Comment by Vampouille** [comment]
Maybe this need some documentation in proxy-permissions.xml in datadir

**Comment by Vampouille** [comment]
> Maybe this need some documentation in proxy-permissions.xml in datadir

Done with https://github.com/georchestra/datadir/pull/60

**Comment by fvanderbiest** [comment]
Denying a range of IPs would be a great addition.

**Comment by fvanderbiest** [comment]
Also requires a comment in the release notes for 17.06

**Comment by Vampouille** [comment]
Check failed on my workstation because some domain resolve has IPv6 address. We need to fix this.

**Comment by Vampouille** [comment]
> Check failed on my workstation because some domain resolve has IPv6 address. We need to fix this.

Should be fixed by 8c7ecfa

**Comment by fvanderbiest** [comment]
Can we merge this now ?

**Comment by Vampouille** [comment]
> Can we merge this now ?

For me, it is ready to be merged.

**Comment by fvanderbiest** [comment]
Runtime error:

18/01/2018 09:55:15org.springframework.beans.factory.parsing.BeanDefinitionParsingException: Configuration problem: You cannot use a spring-security-2.0.xsd or spring-security-3.0.xsd or spring-security-3.1.xsd schema with Spring Security 3.2. Please update your schema declarations to the 3.2 schema.|Offending resource: ServletContext resource [/WEB-INF/applicationContext-security.xml]

https://github.com/georchestra/georchestra/blob/16.12/security-proxy/src/main/webapp/WEB-INF/applicationContext-security.xml#L10 ?

**Comment by fvanderbiest** [comment]
I created issue #1876 to keep a record of this.

**Pull request merged**